js/src/main/kotlin/org/rust/js/RsWasmBindgenLineMarkerProvider.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.js
+
+import com.intellij.codeInsight.daemon.DefaultGutterIconNavigationHandler
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerInfo
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider
+import com.intellij.lang.javascript.psi.JSElement
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptClass
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptEnum
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptFunction
+import com.intellij.navigation.GotoRelatedItem
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.findDescendantOfType
+import icons.JavaScriptPsiIcons
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.psi.impl.RsBaseTypeImpl
+import java.nio.file.InvalidPathException
+import java.nio.file.Paths
+
+class RsWasmBindgenLineMarkerProvider : RelatedItemLineMarkerProvider() {
+    override fun getLineMarkerInfo(element: PsiElement): RelatedItemLineMarkerInfo<*>? {
+        // Line marker will be available only if there is wasm-bindgen dependency in the project,
+        // current element is supported and is public
+        if (element !is RsFunction && element !is RsEnumItem && element !is RsStructItem && element !is RsImplItem) return null
+        if (element is RsElement && element.containingCargoPackage?.dependencies?.none { it.name == "wasm-bindgen" } == true) return null
+        if (element !is RsOuterAttributeOwner && element is RsVisibilityOwner && element !is RsImplItem && !element.isPublic) return null
```

Comment:
Is `element !is RsOuterAttributeOwner` condition reasonable? On this step, we're sure that `element` is a function/enum/struct/impl, and all of them are basically `RsOuterAttributeOwner`

Reply:
I'd recommend to rewrite these checks because now they look a bit confusing
(discussed in person)